samza-core/src/main/java/org/apache/samza/util/DiagnosticsUtil.java

```diff
@@ -101,6 +101,8 @@ public MetadataFileContents(String version, String metricsSnapshot) {
       ClusterManagerConfig clusterManagerConfig = new ClusterManagerConfig(config);
       int containerMemoryMb = clusterManagerConfig.getContainerMemoryMb();
       int containerNumCores = clusterManagerConfig.getNumCores();
+      long maxHeapSizeBytes = Runtime.getRuntime().maxMemory();
+      int containerThreadPoolSize = new JobConfig(config).getThreadPoolSize();
```

Comment:
There is a usage of `JobConfig` above on line 99. Could you please create one `JobConfig` and reuse it?